api/scpca_portal/models/computed_file.py

```diff
@@ -258,7 +257,13 @@ def get_sample_file(
     @property
     def download_url(self):
         """A temporary URL from which the file can be downloaded."""
-        return self.create_download_url()
+        if self.s3_bucket and self.s3_key:
+            # Append the download date to the filename on download.
+            date = utils.get_today_string()
+            s3_key = Path(self.s3_key)
+            filename = f"{s3_key.stem}_{date}{s3_key.suffix}"
+
+            return s3.generate_pre_signed_link(s3_key, filename)
```

Comment:
I'd like to call into question this `if` check.

The old way that we created `ComputedFile` objects was that we would create the object without attributes before archiving and populate the attributes after archiving. In such a scenario, there was a moment that a `ComputedFile` object could be build before having its `s3_bucket` and `s3_key` populated, a relevant possibility that justifies the `if` check.

The current way that we do things is that we first archive the file and then, at completion, we create `ComputedFile` objects with all of their attributes at once. 

As such, this `if` statement does not currently protect from the no longer necessary early accession (i.e. after the object has been created but before the `s3_bucket` and `s3_key` attributes have been added), but rather for the case that these attributes are malformed or empty when their originally passed in. 

Two possible alternatives include:
- Deleting the `if` check all together
- Just checking for these attributes, and if they don't exist then throwing an error or logging it and `return`ing early

Reply:
@davidsmejia